src/lib/promise-all-errors.js

```diff
@@ -0,0 +1,29 @@
+// like Promise.all, but waits for every promise to complete even if one errors
```

Comment:
There is a method for this that will soon be in JavaScript, it is called `Promise.allSettled`.

A polyfill for this is available and works on all NodeJS version you currently support --> https://www.npmjs.com/package/promise.allsettled

The benefit to using the polyfill would be the ability to remove it when this project moves to a version of NodeJS which supports the method natively :-)

Reply:
So that's that all settled then